server/index.js

```diff
@@ -48,13 +47,85 @@ app.get('/api/products/:productId', express.json(), (req, res, next) => {
   db.query(selectId, [id])
     .then(result => {
       if (!result.rows[0]) {
-        return next(new ClientError('The product id does not exist', 404));
+        return next(new ClientError('The productId does not exist', 404));
       }
       res.json(result.rows[0]);
     })
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) return res.json([]);
+  const select = `
+    select "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+     where "c"."cartId" = $1
+  `;
+  db.query(select, [req.session.cartId])
+    .then(result => res.json(result.rows))
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', express.json(), (req, res, next) => {
+  const productId = parseInt(req.body.productId, 10);
+  if (!productId || productId < 1) {
+    return next(new ClientError('The productId must be a postive integer', 400));
+  }
+  const select = `
+    select "price"
+      from "products"
+     where "productId" = $1
+  `;
+  db.query(select, [productId])
+    .then(result => {
+      if (!result.rows[0]) throw next(new ClientError('The productId does not exist', 404));
+      if (req.session.cartId) return req.session.cartId;
+      const insert = `
+        insert into "carts" ("cartId", "createdAt")
+             values (default, default)
+          returning "cartId";
+      `;
+      const newResult = result.rows[0];
+      return db.query(insert)
+        .then(result => Object.assign(newResult, result.rows[0]))
+        .catch(err => next(err));
```

Comment:
Since you have `return` statements in the right places, you don't need to add the `.catch()` here. It can go at the very bottom of your entire promise chain.